src/features/documentSymbolProvider.ts

```diff
@@ -14,43 +14,34 @@ import SymbolRangeNames = protocol.V2.SymbolRangeNames;
 
 export default class OmnisharpDocumentSymbolProvider extends AbstractSupport implements vscode.DocumentSymbolProvider {
 
-    async provideDocumentSymbols(document: vscode.TextDocument, token: vscode.CancellationToken): Promise<vscode.SymbolInformation[]> {
+    async provideDocumentSymbols(document: vscode.TextDocument, token: vscode.CancellationToken): Promise<vscode.DocumentSymbol[]> {
         const response = await serverUtils.codeStructure(this._server, { FileName: document.fileName }, token);
         
         if (response && response.Elements) {
-            return createSymbols(response.Elements, document.fileName);
+            return createSymbols(response.Elements);
         }
 
         return [];
     }
 }
 
-function createSymbols(elements: Structure.CodeElement[], fileName: string): vscode.SymbolInformation[] {
-    let results: vscode.SymbolInformation[] = [];
+function createSymbols(elements: Structure.CodeElement[]): vscode.DocumentSymbol[] {
+    let results: vscode.DocumentSymbol[] = [];
 
-    Structure.walkCodeElements(elements, (element, parentElement) => {
-        const parentElementName = parentElement ? parentElement.DisplayName : undefined;
-        const symbol = createSymbolForElement(element, parentElementName, fileName);
+    Structure.walkCodeElements(elements, (element) => {
+        const symbol = createSymbolForElement(element);
 
         results.push(symbol);
     });
 
     return results;
 }
 
-function createSymbolForElement(element: Structure.CodeElement, parentElementName: string, fileName: string): vscode.SymbolInformation {
-    const range = element.Ranges[SymbolRangeNames.Full];
-
-    const vscodeRange = new vscode.Range(
-        range.Start.Line - 1, range.Start.Column - 1, range.End.Line - 1, range.End.Column - 1
-    );
-
-    return new vscode.SymbolInformation(
-        element.DisplayName,
-        toSymbolKind(element.Kind),
-        parentElementName,
-        new vscode.Location(vscode.Uri.file(fileName), vscodeRange)
-    );
+function createSymbolForElement(element: Structure.CodeElement): vscode.DocumentSymbol {
+    const fullRange = element.Ranges[SymbolRangeNames.Full];
+    const nameRange = element.Ranges[SymbolRangeNames.Name];
+    
+    return new vscode.DocumentSymbol(element.DisplayName, JSON.stringify(element.Properties), toSymbolKind(element.Kind), toVscodeRange(fullRange), toVscodeRange(nameRange));
```

Comment:
Can you share an example of where this gets presented visually so we can see how the detail is shown? I don't think you want a json string.

Reply:
![detail](https://user-images.githubusercontent.com/12554141/45176689-a7ff4c00-b1c5-11e8-89ad-a124ca2967ca.PNG)

The detail is shown in the outline view. And yes you are right we should not put a json string. Should we use the filename instead ?